actions/post_actions.jsx

```diff
@@ -116,31 +115,32 @@ export function unflagPost(postId) {
     };
 }
 
-export async function createPost(post, files, success) {
-    // parse message and emit emoji event
-    const emojis = post.message.match(EMOJI_PATTERN);
-    if (emojis) {
-        for (const emoji of emojis) {
-            const trimmed = emoji.substring(1, emoji.length - 1);
-            dispatch(addRecentEmoji(trimmed));
+export function createPost(post, files) {
+    return async (doDispatch) => {
+        // parse message and emit emoji event
+        const emojis = post.message.match(EMOJI_PATTERN);
+        if (emojis) {
+            for (const emoji of emojis) {
+                const trimmed = emoji.substring(1, emoji.length - 1);
+                dispatch(addRecentEmoji(trimmed));
```

Comment:
This should be `doDispatch` right?

Reply:
actually I'm going to get rid of the global dispatch in this file